### DISCUSSION TITLE
How to convert HostBuffer to String?

### DISCUSSION THREAD

**Discussion by albham** [Moddable-OpenSource/moddable#965] [Q&A] [answered]
Hello,

  I want to modify a webpage content, loaded as a Resource, to change some word 
(in HandleBars style).

So I'm trying to convert the HostBuffer returned by Resource to String so I can do:  `replace('{{ssid}}', ssid)`.

Is this possible? Is there any other better way?

Thanks, Alberto.

**Comment by phoddie** [comment, marked as answer]
A `HostBuffer` can generally be used like an `ArrayBuffer`. You can convert it to a string using the `TextDecoder` API:

```js
import TextDecoder from "text/decoder";

let decoder = new TextDecoder;
let text = decoder.decode(resource);
```

To use `TextDecoder` you will need to add its manifest, `$MODDABLE/modules/data/text/decoder/manifest.json`, to your project's manifest.json.

A lightweight solution is to use the XS API `String.fromArrayBuffer`:

```js
let text = String.fromArrayBuffer(resource);
```

You mention "HandleBars style." It is possible to run Mustache.js in the Moddable SDK. See the [Template node](https://github.com/phoddie/node-red-mcu/tree/main/nodes/template) implementation of Node-RED MCU Edition for an example of that.

**Reply by albham** [reply]
Thank you.

Alberto
